workers/unity/Packages/io.improbable.gdk.testutils/Editor/Tests/SpatialdManagerTests.cs

```diff
@@ -0,0 +1,119 @@
+using System;
+using System.Linq;
+using System.Net;
+using Improbable.Gdk.Tools;
+using NUnit.Framework;
+using UnityEngine;
+
+namespace Improbable.Gdk.TestUtils.Editor.Tests
+{
+    [TestFixture]
+    public class SpatialdManagerTests
+    {
+        [Test]
+        public void Spatiald_is_running_after_Start()
+        {
+            using (SpatialdManager.Start().Result)
+            {
+                var result = RedirectedProcess.Command(Common.SpatialBinary)
+                    .WithArgs("service", "status")
+                    .InDirectory(Common.SpatialProjectRootDir)
+                    .RedirectOutputOptions(OutputRedirectBehaviour.None)
+                    .RunAsync()
+                    .Result;
+
+                Assert.IsTrue(result.Stdout.Any(line => line.Contains("Local API service is running")));
+            }
+        }
+
+        [Test]
+        public void Spatiald_is_not_running_after_Dispose()
+        {
+            using (SpatialdManager.Start().Result)
+            {
+            }
+
+            var result = RedirectedProcess.Command(Common.SpatialBinary)
+                .WithArgs("service", "status")
+                .InDirectory(Common.SpatialProjectRootDir)
+                .RedirectOutputOptions(OutputRedirectBehaviour.None)
+                .RunAsync()
+                .Result;
+
+            Assert.IsTrue(result.Stdout.Any(line => line.Contains("Local API service is not running")));
+        }
+
+        [Test]
+        public void StartLocalDeployment_throws_if_invalid_launch_config_path()
```

Comment:
what happens if the path is fine but the config is invalid?

Reply:
though I suppose it's fine for now since we control the configs for testing

Reply:
ditto for valid snapshot path but invalid snapshot

Reply:
I'd prefer that we don't validate configs / snapshots here as that's not our job. If we aren't getting good error messages back from the deployment launch then its an upstream problem.

I could add tests for this, but it would mean having garbage configs/snaphots committed into the project 😕